Kernel/Syscalls/iwatcher.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/API/InodeWatcherFlags.h>
+#include <Kernel/FileSystem/Custody.h>
+#include <Kernel/FileSystem/FileDescription.h>
+#include <Kernel/FileSystem/InodeWatcher.h>
+#include <Kernel/Process.h>
+
+namespace Kernel {
+
+KResultOr<int> Process::sys$create_iwatcher(u32 flags)
+{
+    REQUIRE_PROMISE(rpath);
+
+    int fd = alloc_fd();
+    if (fd < 0)
+        return fd;
+
+    auto description = FileDescription::create(*InodeWatcher::create());
+    if (description.is_error())
+        return description.error();
+
+    m_fds[fd].set(description.release_value());
+    m_fds[fd].description()->set_readable(true);
+
+    // NOTE: these need to happen after the fd is put in the process fd table.
+    if (flags != 0) {
+        auto flags_or_error = sys$fcntl(fd, F_GETFL, 0);
+        if (flags_or_error.is_error())
+            return flags_or_error.error();
+
+        int fd_flags = flags_or_error.value();
+        if (flags & static_cast<unsigned>(InodeWatcherFlags::Nonblock))
+            fd_flags |= O_NONBLOCK;
+        if (flags & static_cast<unsigned>(InodeWatcherFlags::CloseOnExec))
+            fd_flags |= O_CLOEXEC;
+
+        auto error = sys$fcntl(fd, F_SETFL, fd_flags);
+        if (error.is_error())
+            return error.error();
+    }
```

Comment:
Instead of calling `sys$fcntl()`, just manipulate the flags directly via `m_fds[fd].set_flags()`